src/datasets/other/serp_events/reference.md

```diff
@@ -273,53 +285,43 @@ At the SERP level, ad ad blocker is inferred to be in use if is it inferred on a
-according to the following logic.
-The `ad_blocker_inferred` column indicates whether we assume an ad blocker was in use on a display component.
-
-| `num_ads_hidden_reported` case        | `num_ads_showing`                                      | `num_ads_notshowing`                                                             | `ad_blocker_inferred` |
-| ------------------------------------- | ------------------------------------------------------ | -------------------------------------------------------------------------------- | --------------------- |
-| `== 0`                                | `= num_ads_visible_reported`                           | `= num_ads_loaded_reported - num_ads_visible_reported`                           | `false`               |
-| `== num_ads_loaded_reported`          | `0`                                                    | `0`                                                                              | `true`                |
-| `> 0` and `< num_ads_loaded_reported` | `= num_ads_visible_reported + num_ads_hidden_reported` | `= num_ads_loaded_reported - num_ads_visible_reported - num_ads_hidden_reported` | `false`               |
-
-The last row represents an edge case in which some but not all loaded ads are blocked.
-Per Engineering, this is best interpreted as a shortcoming of the detection method
-rather than an ad blocker operating selectively.
-In this case, ads reported as "blocked" are considered to actually be visible.
+Alongside impression-level fields and summaries,
+the table has 3 array-valued fields:
+
+- `ad_components`, with 1 entry per [ad component](#measurement),
+  listing counts of impressions and engagements
+- `non_ad_impressions`, with 1 entry per non-ad page component,
+  listing impression counts
+- `non_ad_engagements`, with 1 entry per non-ad page component and engagement type,
+  listing engagement counts.
+
+These arrays only include components with at least one non-zero count.
+For SERP impressions with no impressions or engagements,
+the corresponding arrays will be empty.
+
+SERP impressions with no engagements are considered abandoned and have a non-null `abandon_reason`.
+
+### Ad component tagging
+
+Tagging components as containing ads,
+and computing related fields such as `ad_components` and `num_ad_clicks`,
+is implemented at the
+[view layer](https://github.com/mozilla/bigquery-etl/blob/main/sql_generators/serp_events_v2/templates/view.sql)
+using the
+[`is_ad_component` UDF](https://github.com/mozilla/bigquery-etl/blob/main/sql/mozfun/serp_events/is_ad_component/udf.sql).
+
+The underlying
+[derived table](https://github.com/mozilla/bigquery-etl/blob/main/sql_generators/serp_events_v2/templates/desktop_query.sql)
+instead has 2 array-valued fields:
+
+- `component_impressions`, with 1 entry per component reported in a `serp.ad_impression` event
+- `engagements`, with 1 entry per component and engagement action reported in a `serp.engagement` event.
```

Comment:
This might be obvious if the links were working but I don't really understand what this means

Reply:
For each impression we get 0 or more of each of ad_impression and engagement events. For each impression_id, the contents of these multiple events are aggregated into these arrays. The distinction is that the derived table doesn't know about ads vs. non-ads, so it's just impressions and engagements, whereas the view has ads broken out separately.

I can add more about this in the doc if you think it's helpful.

Reply:
Okay that makes more sense.  It couldn't hurt to add a little more detail